olive/model/__init__.py

```diff
@@ -762,6 +762,43 @@ def to_json(self, check_object: bool = False):
         return serialize_to_json(config, check_object)
 
 
+class CompositePyTorchModel(PyTorchModel):
+    def __init__(self, model_components: List[Dict[str, Any]], **kwargs):
+        kwargs = kwargs or {}
+        kwargs["model_file_format"] = ModelFileFormat.PYTORCH_ENTIRE_MODEL
```

Comment:
>ModelFileFormat.PYTORCH_ENTIRE_MODEL [](http://example.com/codeflow?start=38&length=36)

it will override the model_file_format if user specify it explicitly?

Reply:
Can we use a `COMPOSITE_MODEL` for those then? Even if we don't explicity use it now, it can be useful to differentiate between models and model containers.

Reply:
I think it is ok to have a dedicated COMPOSITE_MODEL model file format for these composite models. 
How do you think? @jambayk

Reply:
Yes, that sounds good to me too.